server/src/main/java/keywhiz/service/daos/ClientDAO.java

```diff
@@ -73,6 +74,18 @@ public void deleteClient(Client client) {
     });
   }
 
+  public void sawClient(Client client) {
+    long now = OffsetDateTime.now().toEpochSecond();
```

Comment:
Maybe add a check here: if `now` is within `threshold` seconds of `client.getLastseen()`, skip writing to the database? I think 24h-granularity for last seen is sufficient so threshold could be 24*60*60 which would avoid a lot of writes.